libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,324 @@
+If you're upgrading other products in the stack, also read the
+{stack-ref}/index.html[Elastic Stack Installation and Upgrade Guide]. 
+
+We recommend that you fully upgrade {es} and {kib} to version 7.0
+before upgrading {beats}. If you're on {beats} 6.0 through 6.6,
+upgrade to {beats} 6.7 *before* doing the {es} upgrade.
 
-We recommend that you fully upgrade Elasticsearch and Kibana to version 6.0
-before upgrading Beats. If you are on a Beats 5.x version lower than 5.6, please
-follow the <<upgrading-to-5.6>> section *before* doing the Elasticsearch
-upgrade.
+Upgrading between non-consecutive major versions (e.g. 5.x to 7.x) is not
+supported.
 
-If you're not ready to upgrade Elasticsearch and Kibana to 6.0, that's alright.
-Beats version 6.0 works with Elasticsearch and Kibana version 5.6, so you can
-upgrade Beats now and the rest of the stack later.
+IMPORTANT: Please read through all upgrade steps before proceeding. Some of the
+steps are required before running the upgraded software for the first time.
 
-[[upgrading-to-5.6]]
-==== Upgrade to 5.6
+// REVIEWERS: Historically our docs have not described how to actually upgrade.
+// We just cover the things users need to be aware of. However, to have parity
+// with the install docs for other stack components, I think we should add more
+// detail, similar to what Kibana does: https://www.elastic.co/guide/en/kibana/master/upgrade-standard.html
+// WDYT?
+//
+// Besides copying over the config file (for all beats), and pointing to the 
+// registry file for Filebeat, what else needs to be preserved.
+
+[[upgrading-to-6.7]]
+==== Upgrade to {beats} 6.7 before upgrading to 7.0
 
-The upgrade procedure assumes that you have Beats version 5.6 installed. If you
-are on a previous 5.x version of Beats, please upgrade to version 5.6 first.
-The reason is that the Elasticsearch mapping template in 5.6 was modified to be
-compatible with Elasticsearch 6.0 (by removing the `_all` settings).
+The upgrade procedure assumes that you have {beats} 6.7 installed. If you're on
+a previous 6.x version of {beats}, upgrade to version 6.7 first. This is
+required because the {es} mapping template in 6.7 was modified to be compatible
+with {es} 7.0 (the `_type` setting changed from `doc` to `_doc`).
 
-For this reason, after upgrading to 5.6, you need to make sure that the 5.6
-template is loaded. You can do this by temporarily enabling the
-`output.elasticsearch.template.overwrite` setting, for example with Metricbeat:
+After upgrading to 6.7, make sure the 6.7 template gets loaded. You can do this
+by temporarily enabling the `output.elasticsearch.template.overwrite` setting,
+for example with Metricbeat:
 
 [source,shell]
-------------------------------------------------------------------------------
+----
 metricbeat -e -E output.elasticsearch.template.overwrite=true
-------------------------------------------------------------------------------
+----
```

Comment:
Better run `<beatname> setup --template` for updating the template only.

Note: beats lazily connects and sets up ES. If a beat has no event to publish, it won't update ES.